src/components/ebay-menu-button/component.js

```diff
@@ -11,26 +10,22 @@ module.exports = {
         }
     },
 
-    toggleItemChecked(itemEl) {
-        const itemIndex = indexOf(itemEl.parentNode.children, itemEl);
-        const item = this.state.items[itemIndex];
-        const currentIndex = this.state.items.findIndex(radioItem => radioItem.checked);
+    toggleItemChecked(index, itemEl) {
+        const currentIndex = this.state.checkedItems.findIndex(checked => checked);
 
-        if (this.state.type === 'radio' && itemIndex !== currentIndex) {
-            this.state.items.forEach((eachItem, i) => {
-                eachItem.checked = i === itemIndex;
-            });
-
-            this.setStateDirty('items');
+        if (this.type === 'radio' && index !== currentIndex) {
+            this.setState('checkedItems', this.state.checkedItems.map((item, i) => i === index));
```

Comment:
In Marko 4 I'd prefer to see `this.state.checkedItems = ` instead of `setState`.